src/services/productService.ts

```diff
@@ -1,22 +1,38 @@
 import { CardProduct, Product } from "model/Product";
 import { DataPage } from "model/dataPage";
 
-import { get } from "./api";
+import { get, post, put } from "./api";
+import { ParamsQuery } from "model/paramsQuery";
 
 export const productService = {
-    async getAllProductCards(category: string): Promise<DataPage<CardProduct[]>> {
-        const result = await get(`product/list/${category}`);
+    async getAllProductCards(signal: AbortSignal, params?: ParamsQuery<null>): Promise<DataPage<CardProduct[]>> {
+        const category = params?.path ?? '';
+        const result = await get(`product/list/${category}`, signal);
 
         return result as DataPage<CardProduct[]>;
     },
-    async getNewProductCards(): Promise<CardProduct[]> {
-        const result = await get('product/new');
+    async getNewProductCards(signal: AbortSignal): Promise<CardProduct[]> {
+        const result = await get('product/new', signal);
 
         return result as CardProduct[];
     },
-    async getProductId(id: string): Promise<Product> {
-        const result = await get(`product/${id}`);
+    async getProductId(signal: AbortSignal, params?: ParamsQuery<null>): Promise<Product> {
+        const id = params?.path ?? '';
+        const result = await get(`product/${id}`, signal);
 
         return result;
     },
+    async postProduct(signal: AbortSignal, params?: ParamsQuery<Product>): Promise<boolean> {
+        const body = params?.body ?? {};
+        const result = await post(`product/`, body, signal);
+
+        return result as boolean;
+    },
+    async putProduct(signal: AbortSignal, params?: ParamsQuery<Product>): Promise<boolean> {
+        const body = params?.body ?? {};
+        const id = params?.path ?? '';
```

Comment:
how `id` can be  equal path?, or const `path = params?.path` or `id = params?.id`

Reply:
please pay attention on this kind of cases